nc/include/component/Collider.h

```diff
@@ -11,15 +11,24 @@ namespace nc
         public:
             Collider(EntityHandle handle, Vector3 scale);
             ~Collider();
-            void Update(graphics::FrameManager& frame);
+
+            void UpdateTransformationMatrix();
+            DirectX::FXMMATRIX GetTransformationMatrix() const;
 
             #ifdef NC_EDITOR_ENABLED
+            void UpdateModel(graphics::FrameManager& frame);
             void EditorGuiElement() override;
+            void SetEditorSelection(bool state);
             #endif
 
-        private:
-            Transform* m_transform;
+        protected:
+            DirectX::XMMATRIX m_colliderMatrix;
+            DirectX::FXMMATRIX m_transformMatrix;
             Vector3 m_scale;
+
+            #ifdef NC_EDITOR_ENABLED
             graphics::Model m_model;
```

Comment:
See other comment - I recommend changing this name to m_colliderWireframe or m_colliderWidget or similar.